addons/ColorPreview/ColorPreview.gd

```diff
@@ -17,6 +18,9 @@ func _enter_tree() -> void:
 	# so, to get it consistently, check if it's there whenever the focus changes
 	get_viewport().gui_focus_changed.connect(get_shader_editor_code_edit)
 
+	regex = RegEx.new()
+	regex.compile("(Color\\((?<params1>(?R)*.*?)\\)|Color\\.(?<const>[A-Z_]+)\\b|source_color.*?vec4\\((?<params2>(?R)*.*?)\\))")
```

Comment:
nice, yeah it does seems to work the same. 
readability suffers a lot though, maybe they could just be split in lines and concatenated after `|`
naming could be improved, maybe `color_params` and `vec4_params`

Reply:
Splitting the regex string would be a good idea. If sometimes someone wants to make any changes to the regexes, smaller ones would be easier to handle.

Reply:
the outermost parentheses can also be removed. like this, with new naming
```gdscript
    regex.compile(
        "Color\\((?<color_params>(?R)*.*?)\\)|" +
        "Color\\.(?<color_const>[A-Z_]+)\\b|" +
        "source_color.*?vec4\\((?<vec4_params>(?R)*.*?)\\)"
	)
```
I suppose they could also be in an array again and be `"|".join`ed, but this is absolutely fine